src/Features/Core/Portable/DocumentHighlighting/ExportEmbeddedLanguageBraceMatcherAttribute.cs

```diff
@@ -0,0 +1,68 @@
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using Microsoft.CodeAnalysis.EmbeddedLanguages;
+
+namespace Microsoft.CodeAnalysis.DocumentHighlighting
+{
+    /// <summary>
+    /// Use this attribute to export a <see cref="IEmbeddedLanguageDocumentHighlightsService"/>.
+    /// </summary>
+    [MetadataAttribute]
+    [AttributeUsage(AttributeTargets.Class)]
+    internal class ExportEmbeddedLanguageDocumentHighlightsServiceAttribute : ExportAttribute
+    {
+        /// <summary>
+        /// Name of the classifier.
+        /// </summary>
+        public string Name { get; }
+
+        /// <summary>
+        /// Name of the containing language hosting the embedded language.  e.g. C# or VB.
+        /// </summary>
+        public string Language { get; }
+
+        /// <summary>
+        /// Identifiers in code (or StringSyntaxAttribute) used to identify an embedded language string. For example
+        /// <c>Regex</c> or <c>Json</c>.
+        /// </summary>
+        /// <remarks>This can be used to find usages of an embedded language using a comment marker like <c>//
+        /// lang=regex</c> or passed to a symbol annotated with <c>[StringSyntaxAttribyte("Regex")]</c>.  The identifier
+        /// is case sensitive for the StringSyntaxAttribute, and case insensitive for the comment.
+        /// </remarks>
+        public string[] Identifiers { get; }
+
+        public ExportEmbeddedLanguageDocumentHighlightsServiceAttribute(
+            string name, string language, params string[] identifiers)
+            : base(typeof(IEmbeddedLanguageDocumentHighlightsService))
+        {
+            Name = name ?? throw new ArgumentNullException(nameof(name));
+            Language = language ?? throw new ArgumentNullException(nameof(language));
+            Identifiers = identifiers ?? throw new ArgumentNullException(nameof(identifiers));
+        }
+    }
+
+    /// <summary>
+    /// Internal version of <see cref="ExportEmbeddedLanguageDocumentHighlightsServiceAttribute"/>.  Used so we can allow regex/json
+    /// to still light up on legacy APIs not using the new [StringSyntax] attribute the runtime added.  For public
+    /// extensions that's the only mechanism we support.
+    /// </summary>
+    [MetadataAttribute]
+    [AttributeUsage(AttributeTargets.Class)]
+    internal sealed class ExportEmbeddedLanguageDocumentHighlightsServiceInternalAttribute : ExportEmbeddedLanguageDocumentHighlightsServiceAttribute
+    {
+        /// <inheritdoc cref="EmbeddedLanguageMetadata.SupportsUnannotatedAPIs"/>
+        public bool SupportsUnannotatedAPIs { get; }
+
+        public ExportEmbeddedLanguageDocumentHighlightsServiceInternalAttribute(
+            string name, string language, bool supportsUnannotatedAPIs, params string[] identifiers)
+            : base(name, language, identifiers)
+        {
+            SupportsUnannotatedAPIs = supportsUnannotatedAPIs;
+        }
```

Comment:
at this point, this pattern is repeated 3 times.  strongly thinking of just converting it to two classes which take in the typeof(IEmbeddedLanguageDocumentHighlightsService) etc. data as a parameter at teh use site.

Reply:
Opened https://github.com/dotnet/roslyn/pull/61853 to simplify this.